Added icon to button component, reworked icon display code.

ConorWebb96 opened the pull request:
## Description
This amends the manifest to add an icon selector. Additionally, small code change to allow the displaying of the selected icons.

## Addresses
- [Button enhancement issue](https://github.com/Budibase/budibase/issues/12451)

## App Export
- If possible, attach an app export file along with your request template to make QA testing easier, with minimal setup.

## Screenshots
<img width="260" alt="Screenshot 2023-12-19 at 11 08 12" src="https://github.com/Budibase/budibase/assets/126772285/daf1bb3c-082f-4895-8ad3-1eb46ca6ed44">
<img width="315" alt="Screenshot 2023-12-19 at 11 08 19" src="https://github.com/Budibase/budibase/assets/126772285/f9076194-34de-4dd6-b473-aec87d90fce6">

melohagan reviewed (changes requested):
Pulled it down and it works grand!

I just have a few thoughts:

- Spectrum offer an approach to buttons with icons: https://opensource.adobe.com/spectrum-web-components/components/button/#variants
- There's a comment on this line: https://github.com/Budibase/budibase/blob/c1283860beebabc6e866de38b5c7cd21c17fdb43/packages/client/src/components/app/Button.svelte#L17C1-L17C1 (there may be a reason we don't want to allow icons to be added to buttons, might want to ask AK when he gets back next year)
- This didn't work for buttons within a button group.

aptkingston commented:
Will take a look at this tomorrow @ConorWebb96, sorry it's taken so long. Only just spotted it!

joebudi commented:
From the image of the icon/button, there needs to be more of a gap. Small enhancement, but a big crowd pleaser IMO. Well done @ConorWebb96

ConorWebb96 commented:
Hey @joebudi,

Added the gap setting that should hopefully resolve that spacing issue. Opted to use gap to keep the consistency with the other components that use this.

Did some additional fixes whilst i was in there.
- Allowing to search for icons via hitting enter.
- Added the ability to clear the icon as there wasn't a way to do this before.
- Added gap to button and buttonGroup components.

aptkingston commented on packages/client/src/components/app/Button.svelte:
```diff
@@ -58,15 +59,7 @@
-      >
-        <use xlink:href="#spectrum-icon-18-{icon}" />
-      </svg>
+      <i use:styleable={$component.styles} class="{icon} {size}" />
```
We should remove the `use:styleable` here - that's specifically for the button itself rather than the icon, so could have unintended side effects if also applied here.

ConorWebb96 replied:
Removed 👍

aptkingston reviewed (changes requested):
LGTM other than one small change!

I hate that our icon picker is so terrible though, because this makes it a lot more discoverable. Would be really nice to make a new icon picker first (using spectrum icons) and then use there here, but if this feature is needed in the meantime then all good.

Nice work though @ConorWebb96 - code looks good.

melohagan reviewed (approved):
LGTM! Nice improvements 😄

aptkingston reviewed (approved):
LGTM!
